### PR TITLE
Add FIPS Vendor Affirmation Test for gcrypt

### PR DESCRIPTION
**Pull request by pablo-herranz**
Add a check to verify that the FIPS mode is properly detected by gpgconf

https://progress.opensuse.org/issues/167683

[CI:TOXENVS] fips

### CODE REVIEW

File: tests/test_fips.py [hunk 1]
```diff
@@ -239,12 +239,23 @@ def test_gcrypt_binary(container_per_test: ContainerData) -> None:
 
     """
 
-    container_per_test.connection.check_output(
-        "zypper --gpg-auto-import-keys -n ref && zypper -n in gcc libgcrypt-devel && zypper -n clean && "
+    c = container_per_test.connection
+
+    # Oneline command ready for copy-paste debugging purposes:
+    #   zypper --gpg-auto-import-keys -n ref && zypper -n in gcc libgcrypt-devel && zypper -n clean && gcc -Og -g3 fips-test-gcrypt.c -Wall -Wextra -Wpedantic -lgcrypt -o fips-test-gcrypt && mv fips-test-gcrypt /bin/fips-test-gcrypt
```
Comment (ricardobranco777) [inline tests/test_fips.py:245]: I think we can drop this comment as we can infer this from the commands below.

**Review by ricardobranco777 — APPROVED** [review]
Minor suggestion. Otherwise LGTM.

File: tests/test_fips.py (hunk 1 above)
Comment (pablo-herranz) [reply tests/test_fips.py:245]: Ooops sorry, forget to delete

File: tests/test_fips.py [hunk 2]
```diff
@@ -239,12 +239,21 @@ def test_gcrypt_binary(container_per_test: ContainerData) -> None:
 
     """
 
-    container_per_test.connection.check_output(
-        "zypper --gpg-auto-import-keys -n ref && zypper -n in gcc libgcrypt-devel && zypper -n clean && "
+    c = container_per_test.connection
+
+    c.check_output(
+        "zypper --gpg-auto-import-keys -n ref && "
```
Comment (dcermak) [inline tests/test_fips.py:245]: This shouldn't be necessary
```suggestion
        "zypper -n ref && "
```

File: tests/test_fips.py [hunk 3]
```diff
@@ -239,12 +239,21 @@ def test_gcrypt_binary(container_per_test: ContainerData) -> None:
...
+        "zypper -n in gcc libgcrypt-devel dirmngr && "
+        "zypper -n clean && "
```
Comment (dcermak) [inline tests/test_fips.py:247]: The clean is not necessary, the container is discarded after the test. It just makes the test run longer
```suggestion
```

**Review by grisu48 — APPROVED** [review]

File: tests/test_fips.py (hunk 2 above)
Comment (dcermak) [reply tests/test_fips.py]: Yeah, that was a stupid mistake on my end… https://github.com/SUSE/BCI-tests/pull/697

**Pull request merged**